cpp/include/cudf/copying.hpp

```diff
@@ -14,14 +14,50 @@
  * limitations under the License.
  */
 
-#pragma once 
+#pragma once
 
 #include "cudf.h"
 #include "types.hpp"
+#include <cudf/column/column_view.hpp>
+#include <cudf/table/table.hpp>
+
+#include <memory>
 
 namespace cudf {
 namespace experimental {
 
+/**
+ * @brief Gathers the specified rows (including null values) of a set of columns.
```

Comment:
It should just be implicit that all APIs handle null values from here on out. 
```suggestion
 * @brief Gathers the specified rows of a set of columns according to a gather map. 
```

Reply:
Jake approved but I think this is unaddressed?